tools/publish-aos.js

```diff
@@ -9,46 +12,62 @@ const arweave = Arweave.init({
   port: 443,
   protocol: 'https',
 });
+
 async function main() {
   const bundledLua = fs.readFileSync(
     path.join(__dirname, '../dist/aos-bundled.lua'),
     'utf-8',
   );
-  const wallet = fs.readFileSync(path.join(__dirname, 'key.json'), 'utf-8');
-  const jwk = JSON.parse(wallet);
-  const address = await arweave.wallets.jwkToAddress(jwk);
-
-  const tx1 = await arweave.createTransaction({ data: bundledLua }, jwk);
-  tx1.addTag('App-Name', 'aos-LUA');
-  tx1.addTag('App-Version', '0.0.1');
-  tx1.addTag('Content-Type', 'text/x-lua');
-  tx1.addTag('Author', 'Permanent Data Solutions');
-  await arweave.transactions.sign(tx1, jwk);
-  await arweave.transactions.post(tx1);
-
-  // replace placeholder with actual tx id
-  const bundledLuaWithTxId = bundledLua.replace(srcCodeTxIdPlaceholder, tx1.id);
 
-  console.log(`Publish AOS ANT Lua with address ${address}`);
+  let wallet = process.env.WALLET;
+  if (!wallet) {
+    try {
+      wallet = fs.readFileSync(path.join(__dirname, 'key.json'), 'utf-8');
+    } catch (error) {
+      console.error(
+        'No key found, please set WALLET environment variable or provide a key.json file',
+      );
+      return;
+    }
+  }
+  const jwk = JSON.parse(wallet);
```

Comment:
meh, you can just blow up if we don't have a wallet
```suggestion
const jwk = process.env.WALLET ? JSON.parse(process.env.WALLET) : fs.readFileSync(path.join(__dirname, process.env.WALLET_PATH || 'key.json', 'utf-8');
```

Reply:
just returning here will exit with code 0 - not causing a failure and thus not throw a failure message in slack. hence why i suggested we just throw up on an invalid wallet. please add the suggestion.

cc @atticusofsparta